src/ImathTest/half_c_main.c

```diff
@@ -0,0 +1,224 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenEXR Project.
+//
+
+//#define IMATH_HALF_NO_TABLES_AT_ALL
+//#define IMATH_HALF_EXCEPTIONS_ENABLED
+
+#include <half.h>
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+
+typedef union
+{
+    uint32_t i;
+    float f;
+} c_half_uif;
+
+static const c_half_uif half_to_float[1 << 16] =
+#include "../Imath/toFloat.h"
+
+    static const unsigned short half_eLut[1 << 9] =
+#include "../Imath/eLut.h"
+
+        static short old_convert (int i)
+{
+    int s = (i >> 16) & 0x00008000;
+    int e = ((i >> 23) & 0x000000ff) - (127 - 15);
+    int m = i & 0x007fffff;
+
+    //fprintf( stderr, "old_convert: s %d   e = %d, m = %d\n", s, e, m );
+    if (e <= 0)
+    {
+        if (e < -10)
+        {
+            return s;
+        }
+
+        m = m | 0x00800000;
+
+        int t = 14 - e;
+        int a = (1 << (t - 1)) - 1;
+        int b = (m >> t) & 1;
+
+        m = (m + a + b) >> t;
+
+        //fprintf( stderr, " <OLD> e %d, m: 0x%08X, t %d, a: %d 0x%08X, b: %d 0x%08X\n", e, m, t, a, a, b, b );
+        return s | m;
+    }
+    else if (e == 0xff - (127 - 15))
+    {
+        if (m == 0)
+        {
+            return s | 0x7c00;
+        }
+        else
+        {
+            m >>= 13;
+            return s | 0x7c00 | m | (m == 0);
+        }
+    }
+    else
+    {
+        m = m + 0x00000fff + ((m >> 13) & 1);
+
+        if (m & 0x00800000)
+        {
+            m = 0;  // overflow in significand,
+            e += 1; // adjust exponent
+        }
+        if (e > 30)
+        {
+            return s | 0x7c00; // if this returns, the half becomes an
+        }                      // infinity with the same sign as f.
+        return s | (e << 10) | (m >> 13);
+    }
+}
+
+static uint16_t
+old_method (float f)
```

Comment:
By "old_method" don't you mean "lookuptable_method"? The term "old" will become confusing in the future.